pkg/controller/operators/operatorconditiongenerator_controller.go

```diff
@@ -0,0 +1,203 @@
+type OperatorConditionGenReconciler struct {
+	Client  client.Client
+	factory decorators.OperatorFactory
+	log     logr.Logger
+}
+
+// +kubebuilder:rbac:groups=operators.coreos.com,resources=operatorconditions,verbs=get;list;update;patch;delete
+// +kubebuilder:rbac:groups=operators.coreos.com,resources=operatorconditions/status,verbs=update;patch
+
+// SetupWithManager adds the OperatorCondition Reconciler reconciler to the given controller manager.
+func (r *OperatorConditionGenReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&operatorsv1.Operator{}).
+		Complete(r)
+}
+
+// NewOperatorReconciler constructs and returns an OperatorReconciler.
+// As a side effect, the given scheme has operator discovery types added to it.
+func NewOperatorConditionGenReconciler(cli client.Client, log logr.Logger, scheme *runtime.Scheme) (*OperatorConditionGenReconciler, error) {
+	// Add watched types to scheme.
+	if err := AddToScheme(scheme); err != nil {
+		return nil, err
+	}
+
+	factory, err := decorators.NewSchemedOperatorFactory(scheme)
+	if err != nil {
+		return nil, err
+	}
+
+	return &OperatorConditionGenReconciler{
+		Client:  cli,
+		log:     log,
+		factory: factory,
+	}, nil
+}
+
+// Implement reconcile.Reconciler so the controller can reconcile objects
+var _ reconcile.Reconciler = &OperatorConditionGenReconciler{}
+
+func (r *OperatorConditionGenReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	// Set up a convenient log object so we don't have to type request over and over again
+	log := r.log.WithValues("request", req)
+	log.V(1).Info("reconciling operator")
+
+	in := &operatorsv1.Operator{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, in)
+	if err != nil {
+		log.V(1).Info("Unable to find operator", "error", err)
+		return ctrl.Result{}, err
+	}
+
+	// Wrap with convenience decorator
+	operator, err := r.factory.NewOperator(in)
+	if err != nil {
+		log.Error(err, "Could not wrap Operator with convenience decorator")
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	selector, err := operator.ComponentSelector()
+	if err != nil {
+		return reconcile.Result{Requeue: true}, nil
+	}
+	operatorConditionList := operatorsv1.OperatorConditionList{}
+
+	err = r.Client.List(context.TODO(), &operatorConditionList, &client.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+	if len(operatorConditionList.Items) != 0 {
+		return ctrl.Result{}, nil
+	}
+	r.log.V(1).Info("Creating OperatorCondition")
+
+	// Get CSV Namespace
+	csvList := operatorsv1alpha1.ClusterServiceVersionList{}
+	err = r.Client.List(context.TODO(), &csvList, &client.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// If you cannot identify the CSV for the operator do not proceed.
+	if len(csvList.Items) != 1 {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	csv := csvList.Items[0]
```

Comment:
you should be able to derive CSV -> serviceaccount names and deployments names from existing machinery in the `install` package, right? no need to query the namespace.

Reply:
That's a good idea.

Reply:
Actually - would we be able to get the name of the deployment this way? What happens if a deployment uses a generateName?

Reply:
All deployments in csvs have names given in the template

Reply:
Perfect, will move forward with the suggested approach.

Reply:
The latest version of the OperatorConditionGenerator Controller reconciles CSVs now. The list of ServiceAccounts and Deployments added to the OperatorCondition's Spec is now based on the contents of the CSV.